Source/Extensions/Blazorise.TreeView/TreeView.razor.cs

```diff
@@ -58,6 +57,39 @@ await foreach ( var nodeState in paramNodes.ToNodeStates( HasChildNodesAsync, Ha
                 treeViewNodeStates.Add( nodeState );
             }
         }
+
+        if ( paramNodes is INotifyCollectionChanged observableCollection )
+        {
+            observableCollection.CollectionChanged -= OnCollectionChanged;
+            observableCollection.CollectionChanged += OnCollectionChanged;
+        }
+    }
+
+    private void OnCollectionChanged( object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e )
+    {
+        if ( e.Action == NotifyCollectionChangedAction.Add || e.Action == NotifyCollectionChangedAction.Remove || e.Action == NotifyCollectionChangedAction.Reset )

```

Comment:
My knowledge of `INotifyCollectionChanged` is limited, but should even the `NotifyCollectionChangedAction.Reset` run the same logic? Is that not something like calling .Clear on a List? Or am I wrong?

Reply:
There is `|| e.Action == NotifyCollectionChangedAction.Reset` check also.

Reply:
Clear is still the same as removing elements, just that it removes them all at once. So yes, it should be here.